src/open_inwoner/plans/views.py

```diff
@@ -101,15 +111,16 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         user = self.request.user
         initial_qs = self.get_queryset()
+        args = self.request.GET
         plans = {"extended_plans": False}
 
         # render the extended plan list view when a begeleider is logged in
         if user.contact_type == ContactTypeChoices.begeleider:
             plans["extended_plans"] = True
+            available_contacts = self.get_available_contacts_for_filtering(initial_qs)
 
-            # filter plans if necessary
-            if "plan_contacts" in self.request.GET or "status" in self.request.GET:
-                filtered_plans = self.get_filtered_plans(initial_qs)
+            if "plan_contacts" in args or "status" in args or "query" in args:
+                filtered_plans = self.get_filtered_plans(initial_qs, available_contacts)
```

Comment:
This construction is not great: now both get_context_data() and get_filtered_plans() need to know about these GET arguments. 

So I'd move this decision into get_filtered_plans().